### PR TITLE
Resolved issue  #4   Support bullet lists starting with * or +

### PR DESCRIPTION
**Pull request by maodd**
Nothing special, original state was - syntax only. I added * and + in grammar file, re-run generate command to update token file. Only issue was * conflicting with em1 syntax (** **), trick is to pull list parser before em.

### CODE REVIEW

File: Example/ExampleTests/SnapshotTests.m [hunk 1]
```diff
@@ -49,21 +53,7 @@ - (void)testFontChange {
...
-    shadow.shadowOffset = CGSizeMake(0.5, 0.5);
-    shadow.shadowBlurRadius = 0.5;
-    NSMutableParagraphStyle *paragraphStyle = [[NSMutableParagraphStyle alloc] init];
-    paragraphStyle.lineSpacing = 6;
-    parser.topAttributes = @{NSForegroundColorAttributeName: [UIColor darkGrayColor],
-                             NSShadowAttributeName: shadow,
-                             NSParagraphStyleAttributeName: paragraphStyle};
-    label.attributedText = [parser attributedStringFromMarkdownString:markdown];
-
-    FBSnapshotVerifyView(label, nil);
```
Comment (gskbyte) [inline Example/ExampleTests/SnapshotTests.m]: why have you replaced this test?

**Comment by gskbyte** [comment]
@maodd thanks for your pull request! However, tests are still red, could you please fix them in order to merge your changes? Thanks!

**Comment by gskbyte** [comment]
@maodd it looks the problem is that you are running tests against iOS 9 and the CI runs them on iOS 8.1. Could you please replace it in the Rakefile? Travis should be able to run the tests on iOS 9.3: https://docs.travis-ci.com/user/languages/objective-c/

**Comment by maodd** [comment]
I re did record in iOS 8.1 instead. but seems still couldn't make test happy.

one of the failed test is testHeaders:

screen shot 2016-02-02 at 9 30 43 am
screen shot 2016-02-02 at 9 30 54 am

I just can't tell the difference between those 2 images.

**Comment by maodd** [comment]
test plaintext result: snapshot image area look different. why?

screen shot 2016-02-02 at 9 40 08 am
screen shot 2016-02-02 at 9 40 16 am

**Comment by maodd** [comment]
Really can't tell the difference in testStyles result:

screen shot 2016-02-02 at 9 41 47 am
screen shot 2016-02-02 at 9 41 55 am

**Comment by maodd** [comment]
Font Change results compare:

screen shot 2016-02-02 at 9 43 35 am
screen shot 2016-02-02 at 9 43 42 am

**Comment by maodd** [comment]
test paragraph result compare:

screen shot 2016-02-02 at 9 44 42 am
screen shot 2016-02-02 at 9 44 48 am

**Comment by maodd** [comment]
Good news is 3 of 7 tests passed (including my newly added testLists)
Bad news is the other 4 failed and my eyes can't tell the difference but Kaleido can.

**Comment by maodd** [comment]
OK. all tests passed now. need to record in iPhone5s.

File: src/XNGMarkdownParser.m [hunk 2]
```diff
@@ -319,28 +337,16 @@ - (void)consumeToken:(XNGMarkdownParserCode)token text:(char *)text {
...
-            textAsString = @"•\t";
-            break;
-        }
         case MARKDOWN_NEWLINE: {
             textAsString = @"";
             break;
         }
         case MARKDOWN_URL: {
+            
+            if ([textAsString hasPrefix:@"@"]) {
```
Comment (gskbyte) [inline src/XNGMarkdownParser.m]: what is this for?

**Comment by gskbyte** [comment]
Thanks for fixing the tests @maodd ! I just left a small comment on your code, after you answer we can merge the PR :)

File: src/XNGMarkdownParser.m (hunk 2 above)
Comment (maodd) [inline src/XNGMarkdownParser.m]: this for another issue (#17) reported and I also encountered, domain in email address shouldn't not be rendered as link. I tried to do pure regex fix, no good. ended up with let regex expend optional leading at sign (@) , and let code to skip domain in email.

Comment (gskbyte) [inline src/XNGMarkdownParser.m]: Well, that might be valid as a temporal fix

**Pull request merged**